client/build_prune.go

```diff
@@ -45,21 +51,22 @@ func (cli *Client) BuildCachePrune(ctx context.Context, opts BuildCachePruneOpti
 	}
 	f, err := filters.ToJSON(opts.Filters)
 	if err != nil {
-		return nil, fmt.Errorf("prune could not marshal filters option: %w", err)
+		return out, fmt.Errorf("prune could not marshal filters option: %w", err)
```

Comment:
For these I'm always a bit on the fence; we've had some cases where the return variable was _partially_ propagated, and on errors we would return a partial result; from that perspective, using an explicit return of a zero struct could prevent that (and make it clearer when reading the code that there's "nothing returned");


```suggestion
		return BuildCachePruneResult{}, fmt.Errorf("prune could not marshal filters option: %w", err)
```

Using a pointer value and a `nil` return would of course work, but may be redundant for these (or at least, for small structs) - and (unfortunately?) the Go proposal to add a `zero` builtin didn't make it https://github.com/golang/go/issues/61372 (which would've been nice for these).